coalib/core/CircularDependencyError.py

```diff
@@ -12,8 +13,24 @@ def __init__(self, node=None):
             The node that was encountered twice and closes the dependency
             circle.
         """
-        message = 'Circular dependency detected.'
+        prefix = 'Circular dependency detected'
         if node is not None:
-            message += ' {0!r} -> ... -> {0!r}'.format(node)
+            message = '{prefix}: {node!r} -> ... -> {node!r}'.format(
+                prefix=prefix, node=node)
+        else:
+            message = prefix + '.'
+        return cls(message)
 
-        Exception.__init__(self, message)
+    @classmethod
+    def for_bears(cls, bears):
+        """
+        Creates the CircularDependencyError with a helpful message about the
+        dependency.
+
+        :param bears:
+            sequence of bears, for which a circular dependency was detected.
+        """
+        bear_names = [bear.name for bear in bears]
+
+        return cls('Circular dependency detected: {names}'.format(
+            names=' -> '.join(bear_names)))
```

Comment:
Yes, it's a matter of taste. Regarding the code duplication: This is not really an issue IMO, since in the design of the CircularDependencyError class in this PR, the two class methods could both call a common implementation. For the usecases that may arise in the future, this class probably has to be redesigned anyway.